tephra-core/src/main/java/org/apache/tephra/txprune/TransactionPruningRunnable.java

```diff
@@ -57,8 +60,13 @@ public void run() {
       Transaction tx = txManager.startShort();
       txManager.abort(tx);
 
+      if (tx.getInvalids().length == 0) {
+        LOG.info("Invalid list is empty, not running transaction pruning");
```

Comment:
is this correct? Doesn't pruning also remove stale/out-of-date entries written by the plugins? Doesn't that cleanup have to happen regardless of whether we actually prune the invalid list?